### DISCUSSION TITLE
Id a missing partial paths

### DISCUSSION THREAD

**Discussion by Sim4n6** [github/codeql#12192] [Q&A] [answered]
The predicates `hasPartialFlow()` and `hasPartialFlowRev()` can be used to search for all potential partial flows from the source or to the sink in PartialPath dataflow queries.

But is there a way, to write a dataflow query that identify a partial path from the source to a sink with one or two missing steps? 

I mean something with `Where config.hasPartialFlowRev() and config.hasPartialFlow() and distance<=2`, please?

**Comment by alexet** [comment, marked as answer]
Hi

In general this is not possible. The problem is that there could always be a step that goes from any node reachable from the source to a node reachable in the sink so all places are theoretically reachable in 1 hop.
